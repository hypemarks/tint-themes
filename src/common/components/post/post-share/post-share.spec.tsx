import React from 'react';
import { render } from '@testing-library/react';
import PostShare from './post-share';

describe('Test <PostShare /> Component', () => {
  it('should match snapshot', () => {
    const wrapper = render(<PostShare url={'url'} text={'text'} imageUrl={'image_url'} id={'id'} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
