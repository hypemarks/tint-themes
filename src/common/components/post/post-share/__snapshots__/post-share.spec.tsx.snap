// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PostShare /> Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="post-share"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-share-alt fa-w-14 fa-xs "
      color="#ccc"
      currentitem="false"
      data-event="focus"
      data-event-off=""
      data-for="share--id"
      data-icon="share-alt"
      data-prefix="fas"
      data-tip="true"
      focusable="false"
      role="img"
      tabindex="0"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M352 320c-22.608 0-43.387 7.819-59.79 20.895l-102.486-64.054a96.551 96.551 0 0 0 0-41.683l102.486-64.054C308.613 184.181 329.392 192 352 192c53.019 0 96-42.981 96-96S405.019 0 352 0s-96 42.981-96 96c0 7.158.79 14.13 2.276 20.841L155.79 180.895C139.387 167.819 118.608 160 96 160c-53.019 0-96 42.981-96 96s42.981 96 96 96c22.608 0 43.387-7.819 59.79-20.895l102.486 64.054A96.301 96.301 0 0 0 256 416c0 53.019 42.981 96 96 96s96-42.981 96-96-42.981-96-96-96z"
        fill="currentColor"
      />
    </svg>
    <div
      class="__react_component_tooltip place-top type-dark allow_click"
      data-id="tooltip"
      id="share--id"
    >
      <ul>
        <li
          class="share-option"
        >
          <a
            class="share-option__link"
            href="https://twitter.com/intent/tweet?url=url&text=text"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-twitter fa-w-16 fa-xs "
              color="#00aced"
              data-icon="twitter"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                fill="currentColor"
              />
            </svg>
            <span
              class="share-name"
            >
              Twitter
            </span>
          </a>
        </li>
        <li
          class="share-option"
        >
          <a
            class="share-option__link"
            href="http://www.facebook.com/sharer.php?u=url&via=tint&t=text"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-facebook-f fa-w-10 fa-xs "
              color="#3b5998"
              data-icon="facebook-f"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                fill="currentColor"
              />
            </svg>
            <span
              class="share-name"
            >
              Facebook
            </span>
          </a>
        </li>
        <li
          class="share-option"
        >
          <a
            class="share-option__link"
            href="https://www.linkedin.com/shareArticle?url=url&title=text&source=TINT"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-linkedin-in fa-w-14 fa-xs "
              color="#007bb6"
              data-icon="linkedin-in"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                fill="currentColor"
              />
            </svg>
            <span
              class="share-name"
            >
              LinkedIn
            </span>
          </a>
        </li>
        <li
          class="share-option"
        >
          <a
            class="share-option__link"
            href="https://www.pinterest.com/pin/create/button/?url=url&description=text&media=image_url"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-pinterest fa-w-16 fa-xs "
              color="#cb2027"
              data-icon="pinterest"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 256c0 137-111 248-248 248-25.6 0-50.2-3.9-73.4-11.1 10.1-16.5 25.2-43.5 30.8-65 3-11.6 15.4-59 15.4-59 8.1 15.4 31.7 28.5 56.8 28.5 74.8 0 128.7-68.8 128.7-154.3 0-81.9-66.9-143.2-152.9-143.2-107 0-163.9 71.8-163.9 150.1 0 36.4 19.4 81.7 50.3 96.1 4.7 2.2 7.2 1.2 8.3-3.3.8-3.4 5-20.3 6.9-28.1.6-2.5.3-4.7-1.7-7.1-10.1-12.5-18.3-35.3-18.3-56.6 0-54.7 41.4-107.6 112-107.6 60.9 0 103.6 41.5 103.6 100.9 0 67.1-33.9 113.6-78 113.6-24.3 0-42.6-20.1-36.7-44.8 7-29.5 20.5-61.3 20.5-82.6 0-19-10.2-34.9-31.4-34.9-24.9 0-44.9 25.7-44.9 60.2 0 22 7.4 36.8 7.4 36.8s-24.5 103.8-29 123.2c-5 21.4-3 51.6-.9 71.2C65.4 450.9 0 361.1 0 256 0 119 111 8 248 8s248 111 248 248z"
                fill="currentColor"
              />
            </svg>
            <span
              class="share-name"
            >
              Pinterest
            </span>
          </a>
        </li>
        <li
          class="share-option"
        >
          <a
            class="share-option__link"
            href="mailto:?subject=You+are+going+to+love+this&body=text"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-envelope fa-w-16 fa-xs "
              color="#26E0BD"
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                fill="currentColor"
              />
            </svg>
            <span
              class="share-name"
            >
              Email
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
